frontend/src/components/theme-toggle-button.tsx

```diff
@@ -9,11 +9,13 @@ function ThemeToggle() {
   useLayoutEffect(() => {
     const storedTheme = localStorage.getItem('theme');
     const prefersDark = window.matchMedia?.('(prefers-color-scheme: dark)').matches;
-    setIsDarkTheme(storedTheme === 'dark' || (!storedTheme && prefersDark) || null);
-    if (storedTheme === 'dark' || (!storedTheme && prefersDark)) {
+    const initialTheme = storedTheme === 'dark' || (!storedTheme && prefersDark);
```

Comment:
Can you work on simplifying this, tooo much code cluttered by previous contributor, we need to make it easy to understand.

Reply:
ok Sir